src/autopas/containers/loadEstimators.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file loadEstimators.h
+ *
+ * @date 24 Apr 2020
+ * @author fischerv
+ */
+
+#pragma once
+
+#include <array>
+#include <set>
+#include <vector>
+
+#include "autopas/options/ContainerOption.h"
+#include "autopas/options/LoadEstimatorOption.h"
+#include "autopas/options/TraversalOption.h"
+#include "autopas/utils/ThreeDimensionalMapping.h"
+
+/**
+ * Collection of functions for estimating the load required to update a specific region within a containers
+ */
+namespace autopas::loadEstimators {
+
+/**
+ * Sums up the squared number of particles for all cells within region
+ *
+ * @param cells
+ * @param cellsPerDimension
+ * @param lowerCorner lower boundary indices for region
+ * @param upperCorner upper boundary indices for region
+ * @return estimated load for given region
+ */
+template <class ParticleCell>
+unsigned long squaredParticlesPerCell(const std::vector<ParticleCell> &cells,
+                                      const std::array<unsigned long, 3> &cellsPerDimension,
+                                      const std::array<unsigned long, 3> &lowerCorner,
+                                      const std::array<unsigned long, 3> &upperCorner) {
+  unsigned long sum = 0;
+  for (unsigned long x = lowerCorner[0]; x <= upperCorner[0]; x++) {
+    for (unsigned long y = lowerCorner[1]; y <= upperCorner[1]; y++) {
+      for (unsigned long z = lowerCorner[2]; z <= upperCorner[2]; z++) {
+        auto load =
+            cells[autopas::utils::ThreeDimensionalMapping::threeToOneD(x, y, z, cellsPerDimension)].numParticles();
+        sum += load * load;
+      }
+    }
+  }
+  return sum;
+}
+
+/**
+ * returns set of load estimators compatible with container.
+ *
+ * @param container
+ * @return compatible load estimators
+ */
+static const std::set<autopas::LoadEstimatorOption> allCompatibleLoadEstimators(autopas::ContainerOption container) {
+  switch (container) {
+    case ContainerOption::linkedCells: {
+      return std::set<autopas::LoadEstimatorOption>{LoadEstimatorOption::none,
+                                                    LoadEstimatorOption::squaredParticlesPerCell};
+    }
+    case ContainerOption::verletListsCells: {
+      return std::set<autopas::LoadEstimatorOption>{LoadEstimatorOption::none,
+                                                    LoadEstimatorOption::squaredParticlesPerCell};
+    }
+    default: {
+      return std::set<autopas::LoadEstimatorOption>{};
+    }
+  }
+}
+
+/**
+ * returns whether or not the given traversal uses load estimation.
+ *
+ * @param traversal
+ * @return
+ */
+static const bool usesLoadEstimator(autopas::TraversalOption traversal) {
```

Comment:
to fix the lgtm error, you have to remove the `const` here.
The error arises, as the const has absolutely no effect.
You might, however, replace the `const` with a `constexpr` which tells the compiler that this function could be evaluated at compile time (if the traversal is known at compile time)

Reply:
you are also getting this exact same error from Jenkins (in this case the intel compiler warns about it):
http://vmbungartz10.informatik.tu-muenchen.de/mardyn/job/AutoPas-Multibranch/job/PR-453/10/warnings33Result/